### PR TITLE
`kci user` command for user groups

### PR DESCRIPTION
**Pull request by JenySadadia**
Added commands to `kci user` for getting user groups by ID and attributes.
Newly added sample commands:
```
kci user get_group <group-id>
kci user find_groups
kci user find_groups name=admin
```

### CODE REVIEW

File: kernelci/api/latest.py [hunk 1]
```diff
@@ -89,37 +89,42 @@ def receive_event(self, sub_id: int):
...
-        return self._get(f'node/{node_id}').json()
-
-    def get_nodes(
-        self, attributes: dict,
-        offset: Optional[int] = None, limit: Optional[int] = None
-    ) -> Sequence[dict]:
-        params = attributes.copy() if attributes else {}
-        nodes = []
-
+    def get_api_objs(self, params: dict, path: str, limit: int = None,
```
Comment (gctucker) [inline kernelci/api/latest.py:92]: This seems like it should be a "private" method e.g. `_get_api_objs()`.

File: kernelci/cli/base.py [hunk 2]
```diff
@@ -178,6 +178,11 @@ class Args:  # pylint: disable=too-few-public-methods
         'help': "Path to the dtbs.json file",
     }
 
+    group_id = {
+        'name': '--group-id',
```
Comment (gctucker) [inline kernelci/cli/base.py:182]: How about just using it as a positional argument `group_id` like for `kci node get`?

File: kernelci/cli/user.py [hunk 3]
```diff
@@ -11,6 +11,23 @@
 from .base_api import APICommand
 
 
+class UserAttributesCommand(APICommand):
```
Comment (gctucker) [inline kernelci/cli/user.py:14]: Is this different to how nodes are being searched using arbitrary attributes?  Maybe something more generic for any kind of search (find) query could be used instead.

Comment (JenySadadia) [reply kernelci/cli/user.py]: `pylint` was also complaining about it. Added a generic class in `base_api.py`. Bdw I am thinking of a more suitable name for the class than `AttributesCommand`.

File: kernelci/api/latest.py (hunk 1 above)
Comment (JenySadadia) [reply kernelci/api/latest.py:92]: Oh, yes. I'll fix it.

File: kernelci/cli/base.py (hunk 2 above)
Comment (JenySadadia) [reply kernelci/cli/base.py:182]: Yes, updated it.

**Comment by JenySadadia** [comment]
Tested OK on staging:
```
$ ./kci user find_groups --api-config=staging.kernelci.org
[
{
"id": "649190b01432ccfdb575a250",
"name": "admin"
}
]
```
```
$ ./kci user get_group 649190b01432ccfdb575a250 --api-config=staging.kernelci.org
{
"id": "649190b01432ccfdb575a250",
"name": "admin"
}

```

File: kernelci/cli/base.py [hunk 4]
```diff
@@ -178,6 +178,11 @@ class Args:  # pylint: disable=too-few-public-methods
         'help': "Path to the dtbs.json file",
     }
 
+    group_id = {
+        'name': 'group_id',
+        'help': "User group id",
+    }
+
```
Comment (gctucker) [inline kernelci/cli/base.py:185]: I'm not sure it's useful to have it in the shared class, but that's a grey area anyway.  All the other arguments there are optional ones with `--something`.  This is not a blocking issue but rather something that would need to be revisited once we remove all the deprecated command line tools (or move the code to the `kernelci.legacy` package).

**Review by gctucker — APPROVED** [review]

**Pull request merged**